integration/shareddata/shareddata.go

```diff
@@ -83,6 +84,32 @@ func AllProviders() []Provider {
 	return maps.Values(res)
 }
 
+// Providers are array of providers.
+type Providers []Provider
+
+// Remove returns an array of providers without the specified provider.
+func (ps Providers) Remove(provider string) Providers {
+	var found bool
+	res := make(map[string]Provider, len(ps)-1)
```

Comment:
An alternative way would be to iterate though the slice and call ([`slices.Remove`](https://pkg.go.dev/golang.org/x/exp/slices#Delete)) when the element is found. This way, we don't need to use an exrtra map.

Reply:
Surprisingly this didn't work as straight forward because we alter would alter the caller's slice. I'll keep as it is for now.

Reply:
> because we alter would alter the caller's slice

I didn't understand this part. Does it modify the caller's slice? Or doesn't? I expect the caller's slice to be modified, yes. If we don't want it, we need to make a copy of it.

Reply:
We need to copy it anyway. So I created a new slice.